api-get-object.go

```diff
@@ -105,7 +105,7 @@ func (c Client) GetObject(bucketName, objectName string) (*Object, error) {
 							// Do not set objectInfo from the first readAt request because it will not get
 							// the whole object.
 							reqHeaders.SetRange(req.Offset, req.Offset+int64(len(req.Buffer))-1)
-							httpReader, _, err = c.getObject(bucketName, objectName, reqHeaders)
+							httpReader, objectInfo, err = c.getObject(bucketName, objectName, reqHeaders)
```

Comment:
You need something like a user did following sequence of events tested.

- ReadAt() to read some bytes at an offset.
- Then proceeds to start Read() instead of ReadAt(). 

Or 

- ReadAt() to read some bytes at offset.
- Seek() try to seek past the offset+length used in ReadAt(). 

The reason previously `objectInfo` was not updated in ReadAt() was to ensure that it is updated only when a Read() or a Seek() is called. I am pretty sure setting objectInfo would cause some unknown situations.